### PR TITLE
Add lookup support

### PR DESCRIPTION
**Pull request by derekcaneja**
Add support for checking if a number is a mobile number using twilio lookups api.
Do not send texts to non-mobile numbers.

### CODE REVIEW

File: sms_twilio.js [hunk 1]
```diff
@@ -14,6 +14,59 @@ sms.configure = function (options) {
...
+ * @param user The user to send the link to.
+ *             This should include a first_name and a phone_number.
+ * @param message A message to add to the download text.
+ */
+sms.sendDownloadLink = function (user, message) {
+  if (!user || !isMobileNumber(user.phone_number)) return false;
+
+  var body = 'Welcome to ' + Meteor.settings.APP.NAME + ', ' + user.first_name + '.\n\n' +
+              message + '\n\n' +
+              'Download the app now from http://dispatch.me/get';
```
Comment (jperl) [inline sms_twilio.js]: This is not generic.

Comment (jperl) [inline sms_twilio.js]: I do not think this method belongs in this package.

File: sms_twilio.js [hunk 2]
```diff
@@ -14,6 +14,59 @@ sms.configure = function (options) {
   sms.from = options.from;
 };
 
+var lookups = new Meteor.Collection('meteor_accounts_sms_lookups');
+
+// Check if a number is a mobile number with twilio lookups.
+var isMobileNumber = function (phone) {
```
Comment (jperl) [inline sms_twilio.js]: :+1: Expose this as `Accounts.sms.lookup`

File: sms_twilio.js [hunk 3]
```diff
@@ -14,6 +14,59 @@ sms.configure = function (options) {
...
+  // XXX Use twilio lookups node library
+  var response = HTTP.call('GET', 'https://lookups.twilio.com/v1/PhoneNumbers/' + phone, {
+    auth: Meteor.settings.TWILIO.SID + ':' + Meteor.settings.TWILIO.TOKEN,
+    params: {
+      Type: 'carrier'
+    }
+  });
+
+  if (response.data) {
+    lookups.insert(_.extend(response.data, { timestamp: new Date() }));
```
Comment (jperl) [inline sms_twilio.js]: Please do a check on `response.data` so we can ensure the schema is correct.

Comment (raix) [inline sms_twilio.js]: don't think we should add the `+1` logic in this package - would hinder use of non US phone nr.?

Comment (jperl) [inline sms_twilio.js]: Yeah def should not have the `+1` --- thanks, I missed that.

**Comment by jperl** [comment]
I would change this PR to "Add lookup support"

**Pull request closed without merging**